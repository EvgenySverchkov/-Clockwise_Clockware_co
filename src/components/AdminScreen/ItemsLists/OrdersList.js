import React from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";

import {
  initOrders,
  deleteOrderFromState,
} from "../../../store/adminPanel/orders/actions";

import {changeListIsLoad} from "../../../store/adminPanel/services/actions";
import deleteDataFromServer from "../services/deleteDataFromServer";

import List from "../List";

import { SERVERDOMAIN } from "../../../services/serverUrls";

const OrdersList = ({history}) => {
  const state = useSelector(state=>{
    return {
      ordersArr: state.orders_reducer.ordersArr,
      listIsLoad: state.main_adminPanel_reduser.listIsLoad
    }
  });
  const dispatch = useDispatch();

  function getOrdersFromServerToState() {
    const headers = {
      Authorization: "Bearer " + sessionStorage.getItem("token") || "",
      "Content-Type": "application/json",
    };
    dispatch(changeListIsLoad(true));
    fetch(`${SERVERDOMAIN}/orders`, { headers })
      .then((json) => json.json())
      .then((data) => {
        dispatch(initOrders(data))
        dispatch(changeListIsLoad(false))
      })
      .catch(err=>{
        alert(err)
        dispatch(changeListIsLoad(false))
      });
  }
  function deleteOrderById(orderId) {
    deleteDataFromServer(`${SERVERDOMAIN}/orders/delete/${orderId}`)
      .then((data) => {
        if (data.success) {
          dispatch(deleteOrderFromState(data.payload));
        } else {
          alert(data.msg);
        }
      })
      .catch((err) => alert(err));
  }
  return (
    <List
      dataArr={state.ordersArr}
      deleteAction={deleteOrderById}
      mainRows={["name", "time", "date", "town"]}
      getData={getOrdersFromServerToState}
      history={history}
      listIsLoad = {state.listIsLoad}
    />
  );
};

OrdersList.propTypes = {
  history: PropTypes.object,
};

export default OrdersList;
