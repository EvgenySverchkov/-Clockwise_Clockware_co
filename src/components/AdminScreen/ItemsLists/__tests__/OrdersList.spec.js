import React from "react";
import {create, act} from "react-test-renderer";
import {Provider} from "react-redux";
import mockStoreConfig from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";
   
import OrdersList from "../OrdersList";

const mockStore = mockStoreConfig();

describe("Test of <OrdersList />", ()=>{
    global.fetch = jest.fn(()=>Promise.resolve({json: jest.fn(()=>Promise.resolve())}));
    const store = mockStore({
        ordersReducer: {
            ordersArr: [{
                id: 1,
                name: "Name",
                email: "example@mail.com",
                size: "small",
                town: "Town",
                date: "2021-02-10",
                time: "12:00",
                masterId: 1,
                endTime: "13:00"
            }]
        },
        adminModalWindows: {
            listIsLoad: false
        }
    });
    const mockProps = {
        history: {
            location: "/"
        }
    };

    it("When list is already loaded", async()=>{
        let component;
        await act(async ()=>{
            component = await create(
                <Provider store={store}>
                    <MemoryRouter initialEntries={["/"]}>
                        <OrdersList {...mockProps}/>
                    </MemoryRouter>
                </Provider>
            );
        })
        expect(component.toJSON()).toMatchSnapshot();
    });
    
    it("When list is loading", async()=>{
        const store = mockStore({ordersReducer:{ordersArr: []}, adminModalWindows: {listIsLoad: true}});
        let component;
        await act(async ()=>{
            component = await create(
                <Provider store={store}>
                    <MemoryRouter initialEntries={["/"]}>
                        <OrdersList {...mockProps}/>
                    </MemoryRouter>
                </Provider>
            );
        })
        expect(component.toJSON()).toMatchSnapshot();
    });
});