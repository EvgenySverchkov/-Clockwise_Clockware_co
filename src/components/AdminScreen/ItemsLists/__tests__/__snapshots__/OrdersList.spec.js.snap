// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <OrdersList /> When list is already loaded 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Time
        </th>
        <th>
          Date
        </th>
        <th>
          Town
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          Name
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          12:00
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          2021-02-10
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          Town
        </td>
        <td
          className="text-center"
        >
          <div
            className="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              className="btn btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuButton"
              type="button"
            >
              Options
            </button>
            <div
              aria-labelledby="dropdownMenuLink"
              className="dropdown-menu"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
              >
                Delete
              </button>
              <a
                className="dropdown-item"
                href="/1"
                onClick={[Function]}
              >
                Edit
              </a>
              <button
                className="dropdown-item"
                data-target="#fullInfoModal"
                data-toggle="modal"
                onClick={[Function]}
                type="button"
              >
                Show full info
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test of <OrdersList /> When list is loading 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th>
          Loading...
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
